python/lsst/ts/salobj/topics/read_topic.py

```diff
@@ -479,6 +484,13 @@ def flush(self):
     def get(self, flush=True):
         """Get the most recently seen message, or `None` if no data ever seen.
 
+        This routine is *not* a pop operation (see `next`) but will flush
```

Comment:
I suppose I'm rather confused regarding why the `get` method and the `aget` method differ regarding their behaviour/presence of the `flush` attribute.

It feels like the flush really shouldn't be here, and it appears to be here because of backwards compatibility. Is it getting deprecated soon?

Reply:
Good question.

The `flush` argument for `get` is a holdover from the very early days of salobj. If I was starting over I would either omit, require it, or change the default to False. Given all the existing code the safest thing to do is require it, and https://github.com/lsst-ts/ts_salobj/pull/145 is the first step: issue a deprecation warning if the argument is omitted.

Rather than repeat that mistake for `aget` (a more recent addition) I omitted the `flush` argument.